server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java

```diff
@@ -44,7 +44,10 @@ public class DataLifecycle implements SimpleDiffable<DataLifecycle>, ToXContentO
 
     private static final FeatureFlag DLM_FEATURE_FLAG = new FeatureFlag("dlm");
 
-    public static final DataLifecycle EMPTY = new DataLifecycle();
+    // This represents a lifecycle with infinite retention that will overwrite other retention periods during composition
+    public static final DataLifecycle EXPLICIT_INFINITE_RETENTION = new DataLifecycle(Retention.NULL);
+    // This represents a lifecycle with infinite retention that will inherit other retention periods during composition
+    public static final DataLifecycle IMPLICIT_INFINITE_RETENTION = new DataLifecycle((TimeValue) null);
```

Comment:
I think these could live in the tests (in the fixture perhaps?) as they're only used in tests? 

If we do keep them here I don't think we should mention composition (as the `DataLifecycle` shouldn't know of composition)

Reply:
oh I missed this, definitely, moving the right now.